models/DataObject/ClassDefinition.php

```diff
@@ -504,6 +504,40 @@ public function save($saveDefinitionFile = true)
 
         $cd .= "\n\n";
 
+        $systemFieldDataParamDoc = 'mixed $data';
+        $reflectionMethod = new \ReflectionMethod(DataObject\ClassDefinition\Data::class, 'addListingFilter');
+        if(preg_match('/@param\s+([^\s]+)\s+\$data(.*)/', $reflectionMethod->getDocComment(), $dataParam)) {
+            $systemFieldDataParamDoc = $dataParam[1].' $data '.$dataParam[2];
+        }
+        $systemFieldOperatorParamDoc = 'string $operator SQL comparison operator, e.g. =, <, >= etc. You can use "?" as placeholder, e.g. "IN (?)"';
+        if(preg_match('/@param\s+([^\s]+)\s+\$operator(.*)/', $reflectionMethod->getDocComment(), $dataParam)) {
+            $systemFieldOperatorParamDoc = $dataParam[1].' $operator '.$dataParam[2];
+        }
+
+        foreach(Service::getSystemFields() as $systemField) {
+            if($systemField === 'o_fullpath') {
+                continue;
+            }
+
+            $fieldName = str_replace('o_', '', $systemField);
+
+            $cd .= '/**' . "\n";
+            $cd .= '* Filter by ' . str_replace(['/**', '*/', '//'], '', $fieldName) . " (system field)\n";
+            $cd .= '* @param '.$systemFieldDataParamDoc."\n";
+            $cd .= '* @param '.$systemFieldOperatorParamDoc."\n";
+            $cd .= '* @return static'."\n";
+            $cd .= '*/' . "\n";
+
+            $cd .= 'public function filterBy' . ucfirst($fieldName) .' ($data, $operator = \'=\') {'."\n";
+            $cd .= "\t" . 'if(strpos($operator, \'?\') === false) {'."\n";
+            $cd .= "\t\t".'$operator .= \' ?\';'."\n";
+            $cd .= "\t}\n";
+            $cd .="\t" . '$listing->addConditionParam(\'`'.$systemField.'` \'.$operator, $data);' . "\n";
```

Comment:
`$listing` is undefined in this context. 
I think it would make also sense to move this part
```php
	if(strpos($operator, '?') === false) {
		$operator .= ' ?';
	}
	$listing->addConditionParam('`o_key` '.$operator, $data);
```

to a helper function, so that we can apply changes without regenerating the class files.

Reply:
Is ok now?